exonum/src/node/mod.rs

```diff
@@ -898,34 +901,75 @@ impl Node {
     /// Explorer api prefix is `/api/explorer`
     /// Public api prefix is `/api/services/{service_name}`
     /// Private api prefix is `/api/services/{service_name}`
-    pub fn run(self) -> io::Result<()> {
+    pub fn run(self) -> Result<(), failure::Error> {
         let api_state = self.handler.api_state.clone();
         let blockchain = self.handler.blockchain.clone();
-        let mut api_handlers: Vec<Listening> = Vec::new();
-
-        // Start private api.
-        if let Some(listen_address) = self.api_options.private_api_address {
-            let api_sender = self.channel();
-            let handler = create_private_api_handler(
-                blockchain.clone(),
-                api_state.clone(),
-                api_sender,
-                &self.api_options,
-            );
-            let listener = Iron::new(handler).http(listen_address).unwrap();
-            api_handlers.push(listener);
-
-            info!("Private exonum api started on {}", listen_address);
-        };
 
-        // Start public api.
-        if let Some(listen_address) = self.api_options.public_api_address {
-            let handler = create_public_api_handler(blockchain, api_state, &self.api_options);
-            let listener = Iron::new(handler).http(listen_address).unwrap();
-            api_handlers.push(listener);
+        // Start actix-web api.
+        let api_handlers = self.api_options
+            .private_api_address
+            .map(|addr| (addr, api_ng::ApiScope::Private))
+            .into_iter()
+            .chain(
+                self.api_options
+                    .public_api_address
+                    .map(|addr| (addr, api_ng::ApiScope::Public))
+                    .into_iter(),
+            )
+            .collect::<Vec<_>>();
+        let (actix_sys_tx, actix_sys_rx) = ::std::sync::mpsc::channel();
+        let (api_addr_tx, api_addr_rx) = ::std::sync::mpsc::channel();
+        let api_scopes = api_handlers.clone();
+        let actix_api_thread = thread::spawn(move || -> Result<(), failure::Error> {
```

Comment:
Creating of `actix_api_thread` will look nice in separate method too.

Reply:
Why, instead of methods like `create_web_api` this method invokes only once.

Reply:
Just to get more readable code. It's hard to read big method with a lot of the logic, and we can achieve here something like
```
let handlers = self.get_handlers();
self.start_actix_api(handlers);
self.shutdown();
```

which is pretty straight-forward. Isn't it?